conda/version.py

```diff
@@ -341,7 +341,7 @@ def __new__(cls, spec):
         elif spec.startswith(('=', '<', '>', '!')):
             m = version_relation_re.match(spec)
             if m is None:
-                raise CondaRuntimeError('Invalid version spec: %s' % spec)
+                raise CondaDependencyError('Invalid version spec: %s' % spec)
```

Comment:
Also use this for any place in `Version`, `VersionSpec`, or `MatchSpec` that looks like makes sense.